src/core/lib/security/credentials/external/url_external_account_credentials.cc

```diff
@@ -120,8 +114,19 @@ void UrlExternalAccountCredentials::RetrieveSubjectToken(
   cb_ = cb;
   grpc_httpcli_request request;
   memset(&request, 0, sizeof(grpc_httpcli_request));
-  request.host = const_cast<char*>(url_.authority().c_str());
-  request.http.path = gpr_strdup(url_.path().c_str());
+  absl::StatusOr<URI> tmp_url = URI::Parse(url_);
+  if (!tmp_url.ok()) {
+    FinishRetrieveSubjectToken(
+        "", GRPC_ERROR_CREATE_FROM_COPIED_STRING(
+                absl::StrFormat("Invalid credential source url.").c_str()));
+    return;
+  }
+  URI url_object = *tmp_url;
+  request.host = const_cast<char*>(url_object.authority().c_str());
```

Comment:
I believe this creates an unnecessary copy of the URI. You can treat a StatusOr similar to how you would a smart pointer. `request.host = const_cast<char*>(tmp_url->authority().c_str());`

See https://abseil.io/tips/181.